Esp32 with bluetooth mouse control

irdkwmnsb opened the pull request:
This PR adds the ability to control the cursor position of an iPhone using an esp32 with a bluetooth chip over serial connection. This reduces the delay between user input and actual event on the iphone: we no longer have to wait until the end of the gesture in order to play it.

If we manage to implement grabbing the video with AirServer instead of WDA we might be able to get rid of it entirely without needing to rely on Apple signatures

irdkwmnsb committed:
Initial iMouse PCB support

irdkwmnsb committed:
Add ESP32 bluetooth mouse

irdkwmnsb committed:
Adds device capabilities support

Adds the ability to set and track device capabilities
such as touch and cursor support in the database.

This information is then broadcast via websockets to
clients.

Also adds ESP32 BLE mouse support for iOS devices
and fixes a few bugs related to the BLE connection.

irdkwmnsb committed:
Updates to use ESP32 for touch input

Replaces iMouse with ESP32 for touch input.

This change updates the codebase to use an ESP32 device for touch input, providing a more reliable and configurable solution. It also filters available serial ports to only show ports with Espressif manufacturers.

The changes include:
- Renaming `iMouseHidPath` to `esp32Path` in CLI arguments and configuration.
- Modifying the device discovery to identify ESP32 devices.
- Updating WDA plugin to use `esp32Path` to initilize the ESP32 touch.

irdkwmnsb committed:
Merge branch 'master' into malzhanov/QA-14389/imouse-pcb-support

irdkwmnsb committed:
Clears ESP32 HID path after forking process

Ensures that the `latestEsp32HID` variable is reset after a new process is forked. This prevents potential issues where an outdated or incorrect path might be used in subsequent process creations.

irdkwmnsb committed:
code lint

irdkwmnsb committed:
Improves ESP32 device handling and reconnection

Addresses issues with ESP32 device disconnection and reconnection by:

- Modifies ESP32 device management to handle multiple devices and track their usage.
- Implements a device pool, which enhances the stability of the touch input system.
- Ensures devices are properly returned to the pool upon worker cleanup.
- Adds checks to avoid errors when the ESP32 device is not available.

DaniilSmirnov commented on lib/units/ios-device/plugins/wda.js:
```diff
@@ -11,13 +11,40 @@ import * as iosutil from './util/iosutil.js'
+            dev.on('paired', () => {
+                push.send([
+                    wireutil.global
+                    , wireutil.envelope(new wire.CapabilitiesMessage(options.serial, true, true))
```
Maybe use UpdateDevice message instead of new one? Also we need set to false for other devices

irdkwmnsb replied:
We don't have a generic UpdateDevice message for android and ios
Creating a separate message that will indicate and broadcast what can and cannot do with a device is a good solution by me

DaniilSmirnov commented on lib/units/ios-device/plugins/wda.js:
```diff
@@ -54,14 +81,34 @@ export default syrup.serial()
                 .on(wire.TouchDownMessage, (channel, message) => {
-                    wdaClient.tap(message)
+                    if(dev?.state === 'paired') {
+                        dev.press()
```
maybe device/esp/hid instead of dev?

DaniilSmirnov commented on lib/units/ios-device/plugins/touch/test2.js:
Maybe we need to delete tests?

DaniilSmirnov commented on lib/units/ios-device/plugins/touch/test.js:
Maybe we need to delete tests?

irdkwmnsb committed:
Renames esp32 device variable

Renames the esp32 device variable to better reflect its purpose as a cursor device.

This change improves code readability and clarity by using a more descriptive variable name.

irdkwmnsb committed:
Removes experiment code for iMouse

This commit removes experiment code related to the iMouse implementation.
The deleted files contain testing and debugging code that is no longer needed.
The iMouse class is now initialized to throw an error, indicating that the implementation is not yet complete.

irdkwmnsb committed:
typo
